src/main.nr

```diff
@@ -167,21 +168,39 @@ contract PrivateOracle {
         let address_this = AztecAddress::new(context.this_address());
         let mut escrow_randomness = 0;
 
+ 
+
         if(payment_token.address != 0) {
+
+            let token_contract = TokenPrivateContextInterface::at(payment_token.address);
+
             // Take the payment from the user
-            escrow_randomness = TokenPrivateContextInterface::at(payment_token.address).escrow(
+            escrow_randomness = token_contract.escrow(
                 &mut context,
                 FromEscrowStruct { address: from.address },
-                AgentEscrowStruct { address: address_this.address },
+                OwnerEscrowStruct { address: address_this.address },
                 fee,
-                [
-                    ParticipantsEscrowStruct { address: from.address },
-                    ParticipantsEscrowStruct { address: divinity_address.address },
-                    ParticipantsEscrowStruct { address: 0 },
-                    ParticipantsEscrowStruct { address: 0 }
-                ],
                 nonce
             )[0];
+
+            // Broadcast the escrow to the user
+            let _ = token_contract.broadcast_escrow_note_for(
+                &mut context,
+                AccountBroadcastEscrowNoteForStruct { address: from.address },
+                OwnerBroadcastEscrowNoteForStruct { address: address_this.address },
+                fee,
+                escrow_randomness
+            );
+
+            // Broadcast the escrow to the divinity
+            let _ = token_contract.broadcast_escrow_note_for(
+                &mut context,
+                AccountBroadcastEscrowNoteForStruct { address: divinity_address.address },
+                OwnerBroadcastEscrowNoteForStruct { address: address_this.address },
+                fee,
+                escrow_randomness
+            );
+
```

Comment:
I can't comment in the specific line because this is old, but shouldnt this:

```rs
    // Assert if question doesn't already exist from the same requester
    let question_getter_option = NoteGetterOptions::new().select(0, question).select(1, from.address).set_limit(1);
    let question_note = storage.questions.get_notes(question_getter_option)[0];
```

be checked before all the escrowing logic?

Reply:
There is no "gas"/network cost here, if there is no question, then you cannot generate a proof (in your pxe)

Reply:
Yes, I was referring to saving on computation given that this is processed in the user pxe's right?